### ISSUE TITLE
When creating a new grant, allow ENS names to be entered

### ISSUE THREAD

**Issue by mds1** [gitcoinco/web#8065]
Previously, there were issues where the ENS name wouldn't be resolved quickly enough before submitting the form, and an ENS name ended up in the database. In https://github.com/gitcoinco/web/pull/7471/files, we put in a quick fix which just prevents ENS names from being entered on this form.

We should update the new grants form to allow ENS names, resolve them, and store the resolved address in the database

cc @thelostone-mc

**Comment by apbendi** [comment]
@mds1 I don't know what the current state of this functionality is, but this particular PR you referenced [did allow](https://github.com/gitcoinco/web/pull/7471/files#diff-1150c0efb6ffbf8c1c4de4ed91207ef97e177657e07b3f4f219f4b099625c56eR59) ENS names to be entered, it just resolved them, and then replaced the text in the input. It only prevented the form from being submitted if the ENS name had not yet been resolved/replaced with an address.

**Comment by mds1** [comment]
I think ENS is not currently supported because @thelostone-mc had refactored that page to use Vue and didn't add ENS support back during that refactor. Aditya can you confirm?

**Comment by thelostone-mc** [comment]
Yup i re-added that that check to ensure ENS names are not allowed
